src/repo_add_optionals.rs

```diff
@@ -0,0 +1,116 @@
+use std::process::Command;
+
+#[derive(Debug, Default)]
+pub struct RepoAddArg {
+    chart: String,
+    location: String,
+
+    force_update: Option<bool>,
+    insecure_skip_tls_verify: Option<bool>,
+    no_update: Option<bool>,
+
+    ca_file: Option<String>,
+    cert_file: Option<String>,
+    key_file: Option<String>,
+    password: Option<String>,
+    username: Option<String>
+}
+
+impl RepoAddArg {
+    pub fn new(chart: &str, location: &str) -> RepoAddArg {
+        RepoAddArg{
+            chart: chart.into(),
+            location: location.into(),
+            ..RepoAddArg::default()
+        }
+    }
+
+    pub fn set_force_update<S: Into<bool>>(mut self, value: S) -> Self {
+        self.force_update = Some(value.into());
+        self
+    }
+
+    pub fn set_insecure_skip_tls_verify<S: Into<bool>>(mut self, value: S) -> Self {
+        self.insecure_skip_tls_verify = Some(value.into());
+        self
+    }
+
+    pub fn set_no_update<S: Into<bool>>(mut self, value: S) -> Self {
+        self.no_update = Some(value.into());
+        self
+    }
+
+    pub fn set_ca_file<S: Into<String>>(mut self, value: S) -> Self {
+        self.ca_file = Some(value.into());
+        self
+    }
+
+    pub fn set_cert_file<S: Into<String>>(mut self, value: S) -> Self {
+        self.cert_file = Some(value.into());
+        self
+    }
+
+    pub fn set_key_file<S: Into<String>>(mut self, value: S) -> Self {
+        self.key_file = Some(value.into());
+        self
+    }
+
+    pub fn set_password<S: Into<String>>(mut self, value: S) -> Self {
+        self.password = Some(value.into());
+        self
+    }
+
+    pub fn set_username<S: Into<String>>(mut self, value: S) -> Self {
+        self.username = Some(value.into());
+        self
+    }
+}
+
+impl Into<Command> for RepoAddArg {
```

Comment:
This makes me realize, we should be using `impl From<RepoAddArg> for Command`. It is generally better to implement From than Into. We may want to change this for the other commands as well